File: kotlin/src/main/kotlin/Soa.kt
class Soa(aor: Aor) {
    val ids: Array<Int>
    val selectedItems: Array<Boolean>

    init {
        ids = Array(aor.size) { i ->
            aor.items[i].id
        }
        selectedItems = Array(aor.size) { i ->
            aor.items[i].selected
        }
    }

    fun selected(id: Int): Boolean {
        return selectedItems[ids.indexOf(id)]
    }
}

File: scala/src/main/scala/Soa.scala
package dev.csn

class Soa(aor: Aor) {
  val ids = Array.from(aor.items.map(_.id))
  val selectedItems = Array.from(aor.items.map(_.selected))

  def selected(id: Int) : Boolean =
    selectedItems(ids.indexOf(id))

}
